solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+        govPath[1] = address(networkTokenConverterAnchor(_govToken, registry));
+        govPath[2] = address(_govToken);
+
+        return (paths, amounts, govPath);
+    }
+
+    /**
+     * @dev applies the burn incentive fee on the provided amount and returns the net amount and the fee
+     *
+     * @param amount the network tokens amount
+     *
+     * @return the net amount
+     * @return the fee amount
+     */
+    function applyIncentiveFee(uint256 amount) private view returns (uint256, uint256) {
+        uint256 fee = _burnIncentiveFee;
+        if (fee == 0) {
+            return (amount, 0);
+        }
+
+        uint256 incentiveAmount = Math.min(amount.mul(fee).div(PPM_RESOLUTION), _maxBurnIncentiveFeeAmount);
+
+        return (amount.sub(incentiveAmount), incentiveAmount);
+    }
+
+    /**
+     * @dev finds the converter anchor of the 50/50 standard pool converter between the specified token and the network token
+     *
+     * @param token the source token
+     * @param converterRegistry the converter registry
+     *
+     * @return the converter anchor of the 50/50 standard pool converter between the specified token
+     */
+    function networkTokenConverterAnchor(IERC20 token, IConverterRegistry converterRegistry)
+        private
+        view
+        returns (IConverterAnchor)
+    {
+        // initialize both the source and the target tokens
+        IERC20[] memory reserveTokens = new IERC20[](2);
+        reserveTokens[0] = _networkToken;
+        reserveTokens[1] = token;
+
+        // make sure to only look up for 50/50 converters
+        uint32[] memory standardReserveWeights = new uint32[](2);
+        standardReserveWeights[0] = STANDARD_POOL_RESERVE_WEIGHT;
+        standardReserveWeights[1] = STANDARD_POOL_RESERVE_WEIGHT;
+
+        // find the standard pool converter between the specified token and the network token
+        IConverterAnchor anchor =
+            converterRegistry.getLiquidityPoolByConfig(
+                STANDARD_POOL_CONVERTER_TYPE,
+                reserveTokens,
+                standardReserveWeights
+            );
+        require(address(anchor) != address(0), "ERR_INVALID_RESERVE_TOKEN");
+
+        return anchor;
+    }
+
+    /**
+     * @dev ensures that the network is able to pull the tokens from this contact
+     *
+     * @param token the source token
+     * @param network the address of the network contract
+     * @param amount the token amount to approve
+     */
+    function ensureAllowance(
+        IERC20 token,
+        BancorNetwork network,
+        uint256 amount
+    ) private {
+        address networkAddress = address(network);
+        uint256 allowance = token.allowance(address(this), networkAddress);
+        if (allowance < amount) {
+            if (allowance > 0) {
+                token.safeApprove(networkAddress, 0);
+            }
+            token.safeApprove(networkAddress, amount);
+        }
+    }
```

Comment:
I think it's time for us to consider adding this function in Utils.

Reply:
Probably, but we currently don't have a place for that. I'll open a task for it